### PR TITLE
Use relative URLs for client-server communication

### PR DESCRIPTION
**Pull request by vouillon**

### CODE REVIEW

File: src/lib/eliom_request.client.ml [hunk 1]
```diff
@@ -199,6 +199,7 @@ let send ?(expecting_process_page = false) ?cookies_info
       else true
     in
     try_lwt
+Format.eprintf "XXXX %s@." (Url.encode_arguments get_args);
```
Comment (Drup) [inline src/lib/eliom_request.client.ml]: Hum ?

**Comment by Drup** [comment]
Why where we using absolute URLs before ?

**Comment by vasilisp** [comment]
Does this fix any issues (with redirects maybe) when behind a proxy?

I have no comments on the code.

**Comment by vouillon** [comment]
This make Eliom work behing a proxy that does TLS termination. Otherwise, the server returns https URLs instead of http URLs. Some proxys set a ```X-Forwarded-Proto``` header, which is recognized by the accesscontrol Ocsigen extension. But Cloudfront, for instance, sets ```Cloudfront-Forwarded-Proto``` instead...

This also makes it possible to rewrite URLs between the server and the client without having Eliom aware of that.

**Comment by Drup** [comment]
I see. :+1: then :)

**Pull request merged**